src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -221,14 +207,16 @@ void MulticopterPositionControl::Run()
 		return;
 	}
 
-	perf_begin(_cycle_perf);
+	// reschedule backup
+	ScheduleDelayed(100_ms);
 
-	if (_local_pos_sub.update(&_local_pos)) {
+	parameters_update(false);
 
-		poll_subscriptions();
-		parameters_update(false);
+	perf_begin(_cycle_perf);
```

Comment:
I could go either way. I like to use them to get a rough idea for how expensive the core computation is (including worst case). The parameter update is an expensive exception that skews the results.

Long term I'd like to be gathering this data at the PX4 Work Queue level instead and then get rid of a large portion of the perf counters.

Reply:
> The parameter update is an expensive exception that skews the results.

Makes sense. It renders your worst-case time useless for analyzing the worst case in air computation time which usually have no parameter update. I was just curious and though it might be unwanted. ✔️